core/icons/comment_icon.ts

```diff
@@ -218,7 +231,19 @@ export class CommentIcon extends Icon implements IHasBubble, ISerializable {
    */
   onTextChange(): void {
     if (this.textInputBubble) {
-      this.text = this.textInputBubble.getText();
+      const newText = this.textInputBubble.getText();
+      if (this.textInputBubble.getText() !== this.text) {
+        eventUtils.fire(
+          new (eventUtils.get(eventUtils.BLOCK_CHANGE))(
+            this.sourceBlock,
+            'comment',
+            null,
+            this.text,
+            newText,
+          ),
+        );
+      }
+      this.text = newText;
```

Comment:
This can go inside of the above `if`. No reason to set it if it's allready the same!

It might be mor readable if you restructure this to use early returns.

```
if (!this.textInputBubble) return;

const newText = this.textInputBubble.getText();
if (this.text === newText) return;

eventUtils.fire(
  new (eventUtils.get(eventUtils.BLOCK_CHANGE))(
    this.sourceBlock,
    'comment',
    null,
    this.text,
    newText,
  ),
);
this.text = newText;
```